modules/seqexec/server/src/main/scala/seqexec/server/niri/NiriControllerEpics.scala

```diff
@@ -300,7 +300,8 @@ object NiriControllerEpics extends NiriController {
   override def observeProgress(total: Time): fs2.Stream[IO, Progress] =
     ProgressUtil.countdown[IO](total, 0.seconds)
 
-  override def calcTotalExposureTime(cfg: DCConfig): Time = {
+  override def calcTotalExposureTime(cfg: DCConfig): IO[Time] = IO {
```

Comment:
Niri reads an epics channel to calculate the exposure time

Reply:
We come up with some kind of planned observing time estimate in the program model, but I suppose this is more accurate?

Reply:
I'm not totally sure but I think in the Niri case there are some values calculated in the instrument that need to be taken into account